addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -291,6 +291,7 @@ class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
     requisition_id = fields.Many2one('purchase.requisition', string='Purchase Agreement', copy=False)
+    requisition_type = fields.Many2one(related='requisition_id.type_id')
```

Comment:
Could you do a related to requisition_id.is_quantity_copy instead and put readonly on partner_id if is_quantity_copy = 'none' (corresponding to blanket order).
As you can create any requisition_type you want, your attrs is not complete